src/printer.js

```diff
@@ -58,13 +58,29 @@ function hasPrettierIgnoreComment(node) {
   );
 }
 
+function isOutsideRange(node, rangeStart, rangeEnd) {
+  if (!node) {
+    return false;
+  }
+  if (node.type === 'File' || node.type === 'Program') {
+    return false;
+  }
+  const nodeStart = util.locStart(node);
+  const nodeEnd = util.locEnd(node);
+  const isOutside = nodeEnd < rangeStart || nodeStart > rangeEnd;
+  return isOutside;
+}
+
 function genericPrint(path, options, printPath, args) {
   assert.ok(path instanceof FastPath);
 
   var node = path.getValue();
 
   // Escape hatch
-  if (hasPrettierIgnoreComment(node)) {
+  if (
+    hasPrettierIgnoreComment(node) ||
+    isOutsideRange(node, options.rangeStart, options.rangeEnd)
```

Comment:
Actually, it's a little more subtle than that. For example, try formatting the following code with `prettier --range-start 25 --range-end 45`:

```js
if ( {a:1,
b:2} ) {
    tooMuchIndentation();
    // THE ABOVE LINE IS THE ONLY ONE IN THE RANGE 25-45
}
```

The result is:

```js
if ({a:1,
b:2}) {
  tooMuchIndentation();
  // THE ABOVE LINE IS THE ONLY ONE IN THE RANGE 25-45
}
```

which still removes the whitespace around the object literal, but isn't completely formatting it, as a range-less `prettier` invocation would:

```js
if (
  {
    a: 1,
    b: 2
  }
) {
  tooMuchIndentation();
  // THE ABOVE LINE IS THE ONLY ONE IN THE RANGE 25-45
}
```